vm/isolated/isolated.go

```diff
@@ -189,39 +199,64 @@ func (inst *instance) ssh(command string) error {
 	return nil
 }
 
+func (inst *instance) waitRebootAndSSH(rebootTimeout, sshTimeout time.Duration) error {
+	if err := inst.waitForReboot(5 * 60); err != nil {
+		log.Logf(2, "isolated: machine did not reboot")
+		return err
+	}
+	log.Logf(2, "isolated: rebooted wait for comeback")
+	if err := inst.waitForSSH(30 * time.Minute); err != nil {
+		log.Logf(2, "isolated: machine did not comeback")
+		return err
+	}
+	log.Logf(2, "isolated: reboot succeeded")
+	return nil
+}
+
+func (inst *instance) sshArgsEnv() []string {
+	env := os.Environ()
+	args := append(vmimpl.SSHArgs(inst.debug, inst.sshKey, inst.targetPort), inst.sshUser+"@"+inst.targetAddr)
+	env = append(env, fmt.Sprintf("SSHARGS=%v", strings.Join(args, " ")))
+	return env
+}
+
 func (inst *instance) repair() error {
 	log.Logf(2, "isolated: trying to ssh")
-	if err := inst.waitForSSH(30 * time.Minute); err == nil {
-		if inst.cfg.TargetReboot {
-			if len(inst.cfg.USBDevNums) > 0 {
-				log.Logf(2, "isolated: trying to reboot by USB authorization")
-				usbAuth := fmt.Sprintf("%s%s%s", "/sys/bus/usb/devices/", inst.cfg.USBDevNums[inst.index], "/authorized")
-				if err := ioutil.WriteFile(usbAuth, []byte("0"), 0); err != nil {
-					log.Logf(2, "isolated: failed to turn off the device")
-					return err
-				}
-				if err := ioutil.WriteFile(usbAuth, []byte("1"), 0); err != nil {
-					log.Logf(2, "isolated: failed to turn on the device")
-					return err
-				}
-			} else {
-				log.Logf(2, "isolated: ssh succeeded, trying to reboot by ssh")
-				inst.ssh("reboot") // reboot will return an error, ignore it
+	if err := inst.waitForSSH(30 * time.Minute); err != nil {
+		log.Logf(2, "isolated: ssh failed")
+		return fmt.Errorf("SSH failed")
+	}
+	if inst.cfg.TargetReboot {
+		if len(inst.cfg.USBDevNums) > 0 {
+			log.Logf(2, "isolated: trying to reboot by USB authorization")
+			usbAuth := fmt.Sprintf("%s%s%s", "/sys/bus/usb/devices/", inst.cfg.USBDevNums[inst.index], "/authorized")
+			if err := ioutil.WriteFile(usbAuth, []byte("0"), 0); err != nil {
+				log.Logf(2, "isolated: failed to turn off the device")
+				return err
 			}
+			if err := ioutil.WriteFile(usbAuth, []byte("1"), 0); err != nil {
+				log.Logf(2, "isolated: failed to turn on the device")
+				return err
+			}
+		} else {
+			log.Logf(2, "isolated: ssh succeeded, trying to reboot by ssh")
+			inst.ssh("reboot") // reboot will return an error, ignore it
 		}
-		if err := inst.waitForReboot(5 * 60); err != nil {
-			log.Logf(2, "isolated: machine did not reboot")
-			return err
-		}
-		log.Logf(2, "isolated: rebooted wait for comeback")
-		if err := inst.waitForSSH(30 * time.Minute); err != nil {
-			log.Logf(0, "isolated: machine did not comeback")
-			return err
+	}
+	if err := inst.waitRebootAndSSH(5*60, 30*time.Minute); err != nil {
```

Comment:
The first argument is time.Duration, so this 5*60 is 300 nanoseconds. I guess it's not what you meant :)

Reply:
Yikes, thank you, I've fixed it!